src/tools/fuzzing/fuzzing.cpp

```diff
@@ -3418,23 +3462,54 @@ Type TranslateToFuzzReader::getSubType(Type type) {
   } else if (type.isRef()) {
     auto heapType = getSubType(type.getHeapType());
     auto nullability = getSubType(type.getNullability());
+    auto subType = Type(heapType, nullability);
     // We don't want to emit lots of uninhabitable types like (ref none), so
     // avoid them with high probability. Specifically, if the original type was
     // inhabitable then return that; avoid adding more uninhabitability.
-    auto uninhabitable = nullability == NonNullable && heapType.isBottom();
-    auto originalUninhabitable =
-      type.isNonNullable() && type.getHeapType().isBottom();
-    if (uninhabitable && !originalUninhabitable && !oneIn(20)) {
+    if (isUninhabitable(subType) && !isUninhabitable(type) && !oneIn(20)) {
       return type;
     }
-    return Type(heapType, nullability);
+    return subType;
   } else {
     // This is an MVP type without subtypes.
     assert(type.isBasic());
     return type;
   }
 }
 
+Nullability TranslateToFuzzReader::getSuperType(Nullability nullability) {
+  if (nullability == Nullable) {
+    return Nullable;
+  }
+  return getNullability();
+}
+
+HeapType TranslateToFuzzReader::getSuperType(HeapType type) {
```

Comment:
This could also include basic supertypes like `eq` and `any`, etc.

Reply:
I think those already get enough representation from `getReferenceType()` and friends. Though I agree it's not ideal to not do it here, as it seems natural to add them. But the bigger picture I think is that it could be nice to refactor so that `HeapType::getSuperType()` returns basic types too, which would make this just work as expected. That's a longer-term refactoring though as code depends on it not doing that atm...

Reply:
`ref.cast` of a basic heap type is sufficiently different and interesting compared to `ref.cast` of user defined types that it seems worth fuzzing, even basic heap types are well-represented elsewhere already.